lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+          _.defaultsDeep(updateBodyClone, document._source);
 
-        _.defaultsDeep(updateBodyClone, document._source);
+          return resolve(updateBodyClone);
+        });
+      }
+      else {
+        return resolve(request.input.body);
+      }
+    })
+      .then(documentBody => {
+        const errorMessages = verbose ? {} : [];
+        let isValid = true;
+
+        if (collectionSpec) {
+          if (collectionSpec.fields && collectionSpec.fields.children) {
+            try {
+              isValid = isValid && this.recurseFieldValidation(documentBody, collectionSpec.fields.children, collectionSpec.strict, errorMessages, verbose);
+            }
+            catch (error) {
+              if (error.message === 'strictness') {
+                // The strictness message can be received here only if it happens at the validation of the document's root
+                manageErrorMessage('document', errorMessages, 'The document validation is strict; it can not add unspecified sub-fields.', verbose);
+                isValid = false;
+              }
+              else {
+                throw error;
+              }
+            }
+          }
 
-        return resolve(updateBodyClone);
-      });
-    }
-    else {
-      return resolve(request.input.body);
-    }
-  })
-  .then(documentBody => {
-    var
-      isValid = true,
-      errorMessages = verbose ? {} : [];
+          if (collectionSpec.validators) {
+            const filters = this.dsl.test(request.input.resource.index, request.input.resource.collection, documentBody, id);
 
-    if (collectionSpec) {
-      if (collectionSpec.fields && collectionSpec.fields.children) {
-        try {
-          isValid = isValid && this.recurseFieldValidation(documentBody, collectionSpec.fields.children, collectionSpec.strict, errorMessages, verbose);
-        }
-        catch (error) {
-          if (error.message === 'strictness') {
-            // The strictness message can be received here only if it happens at the validation of the document's root
-            manageErrorMessage('document', errorMessages, 'The document validation is strict; it can not add unspecified sub-fields.', verbose);
-            isValid = false;
+            if (!filters.length > 0 || filters[0] !== collectionSpec.validators) {
+              isValid = false;
+              manageErrorMessage('document', errorMessages, 'The document does not match validation filters.', verbose);
+            }
           }
-          else {
-            throw error;
+        }
+
+        return {
+          errorMessages,
+          valid: isValid
+        };
+      })
+      .then(validationObject => {
+        if (!verbose) {
+          // We only modify the request if the validation succeeds
+          if (collectionSpec.fields && collectionSpec.fields.children) {
+            request.input.body = this.recurseApplyDefault(isUpdate, request.input.body, collectionSpec.fields.children);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 120 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)